### DISCUSSION TITLE
Guidelines for changing hostname of RabbitMQ node

### DISCUSSION THREAD

**Discussion by arshad01** [rabbitmq/rabbitmq-server#12288] [Questions] [answered]
### Community Support Policy

- [X] I have read [RabbitMQ's Community Support Policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md)

### RabbitMQ version used

3.13.7

### Erlang version used

26.2.x

### Operating system (distribution) used

Rocky8

### How is RabbitMQ deployed?

RPM package

### rabbitmq-diagnostics status output

See https://www.rabbitmq.com/docs/cli to learn how to use rabbitmq-diagnostics

### Logs from node 1 (with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs

### Logs from node 2 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs

### Logs from node 3 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs

### rabbitmq.conf

```
listeners.tcp.1 = 127.0.0.1:5672
stomp.listeners.tcp.1 = 127.0.0.1:61613
web_stomp.tcp.listener = 127.0.0.1:15674
stomp.proxy_protocol = true
classic_queue.default_version = 2
deprecated_features.permit.management_metrics_collection = false
```

### Steps to deploy RabbitMQ cluster

Single node deployment with classic queues.

### Steps to reproduce the behavior in question

N/A

What are the guidelines for changing hostname of a RabbitMQ node (no cluster) so that all exchanges/queues and data in mnesia is migrated from old to new node name.

Thanks

### advanced.config

See https://www.rabbitmq.com/docs/configure#config-location to find advanced.config file location

### Application code

```python
# relevant messaging-related parts of your code go here
```

### Kubernetes deployment file

```yaml
# Relevant parts of K8S deployment that demonstrate how RabbitMQ is deployed
```

**Comment by michaelklishin** [comment, marked as answer]
This sounds like the same question as https://github.com/rabbitmq/rabbitmq-server/discussions/12289.

You have several options:

 * [Override node's data directory location](https://www.rabbitmq.com/docs/relocate). This won't update queue identities in the database but you can re-declare the queues, and if all queues in the system are transient, you won't have anything to update
 * [Export and import definitions](https://www.rabbitmq.com/docs/definitions) and import them into newly started blank node. This won't migrate the messages but can be good enough
 * Finally, the most complex migration option which migrates everything is [Blue-Green deployment](https://www.rabbitmq.com/docs/blue-green-upgrade) (it assumes that the original node can be started)
 
 There used to be a command that updated the hostname part of internal queue identities but it was removed because it is not compatible with quorum queues and streams.

**Reply by arshad01** [reply]
Thanks @michaelklishin for the reply.
